app/models/targeted_user_link.rb

```diff
@@ -0,0 +1,33 @@
+# == Schema Information
+#
+# Table name: targeted_user_links
+#
+#  id                :uuid             not null, primary key
+#  target_context    :integer
+#  target_model_type :string
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#  target_model_id   :bigint
+#  user_id           :bigint
+#
+class TargetedUserLink < ApplicationRecord
+  belongs_to :user
+  belongs_to :target_model, polymorphic: true, optional: false
+
+  enum target_context: { :avis => 0 }
+  validates :target_context, inclusion: { in: target_contexts }
+
+  def invalid_signed_in_user?(signed_in_user)
+    signed_in_user && signed_in_user != self.user
```

Comment:
```suggestion
    signed_in_user != self.user
```

il me semble que tu as le not nil compris dedans